lib/lotus/model/adapters/sql_adapter.rb

```diff
@@ -281,6 +281,28 @@ def disconnect
 
         private
 
+        # Returns a hash of attributes and their default values
+        #
+        # @api private
+        # @since x.x.x
+        def defaults(collection)
```

Comment:
This logic can be a private method of `Lotus::Model::Adapters::Sql::Collection`. This is a delegator around a Sequel dataset, so you have direct access to schema informations (via `#schema`).